webapp/src/Controller/Jury/ProblemController.php

```diff
@@ -717,6 +717,16 @@ public function testcasesAction(Request $request, int $probId) : Response
             return $this->redirectToRoute('jury_problem_testcases', ['probId' => $probId]);
         }
 
+        $known_md5s = [];
+        foreach ($testcases as $rank => $testcase) {
+            $input_md5 = $testcase->getMd5sumInput();
+            if (isset($known_md5s[$input_md5])) {
+                $this->addFlash('warning',
+                    "Testcase #" . $rank . " has identical input to testcase #" . $known_md5s[$input_md5] . '.');
+            }
+            $known_md5s[$input_md5] = $rank;
```

Comment:
If you have triple testcases then it will only show the last one. Acceptable or make known_md5s[$input_md5] a list of ranks?

Reply:
Check this situation:
![identical](https://user-images.githubusercontent.com/418721/130354934-2294b244-c391-4739-bdae-7e8c8cadbd62.png)

first three testcases are identical and the last two are as well. I think the code is simpler this way but still everything is highlighted. It should be a rare case anyway.

Reply:
Fair enough. :+1: